src/restruct_poc/solver/populate_sparse_row_ptrs_col_inds_constraints.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <Kokkos_Core.hpp>
+
+#include "constraints.hpp"
+
+#include "src/restruct_poc/types.hpp"
+
+namespace openturbine {
+
+struct PopulateSparseRowPtrsColInds_Constraints {
```

Comment:
I like to write the full words in class names, but that's my personal taste.

`PopulateSparseRowPointersColumnIndicesForConstraints`

Reply:
I agree that these sparse kernels could be named better, possibly like `BuildConstraintSparseIndices`. I was following the existing naming. Might be a good thing to keep in mind when cleaning up since there are four such functions.

Reply:
On one hand: yes, the naming is bad.  On the other hand, I didn't work too hard on naming because they will probably need to be reworked/rewritten a few times as we bang them into shape.  

Not only is the accuracy of the sparsity pattern for different use cases a concern, but we will also likely want the pattern to be built in parallel, which will make a difference for large problems (surprising, but I've seen it in my testing), but is non-trivial to implement.  

All of this is to say that spending too much time tweaking the naming of these kernels might not be a great use of time right now.